### PR TITLE
chore: In-process verdaccio server

### PR DESCRIPTION
**Pull request by mjameswh**
## What was changed
- Extract docs build out of main build job
- [publish-to-verdacio]: Use an in-process verdaccio server rather than starting a new one using npx
- [init-to-verdacio]: Configure npm to go through verdaccio only for `@temporalio` packages
- Make the whole process more verbose

## Why
- Fixes some observed CI test flakes
- Make CI tests a little bit faster

### CODE REVIEW

File: scripts/registry.js [hunk 1]
```diff
@@ -1,88 +1,94 @@
...
-            contents = `Error ${e}`;
-          }
-          reject(new Error(`Timed out waiting for verdaccio - ${contents}`));
-        }, 60 * 1000);
-      });
-    } finally {
-      tail.unwatch();
-    }
+
+    app.listen(4873);
```
Comment (bergundy) [inline scripts/registry.js:79]: Is there a way to wait for the server to listen?

File: scripts/utils.js [hunk 2]
```diff
@@ -41,31 +41,18 @@ async function kill(child, signal = 'SIGINT') {
...
-
 function spawnNpxSync(args, opts) {
-  let cmd = 'npx';
+  let fullCommand = ['npm', 'exec', ...args];
 
   // NPX is a .cmd on Windows
   if (process.platform == 'win32') {
-    args = ['/C', cmd, ...args];
-    cmd = 'cmd';
+    fullCommand = ['cmd', '/C', fullCommand];
```
Comment (bergundy) [inline scripts/utils.js:49]:
```suggestion
    fullCommand = ['cmd', '/C', ...fullCommand];
```

File: scripts/registry.js (hunk 1 above)
Comment (mjameswh) [reply scripts/registry.js:79]: I think this should do it. `app` is actually an instance of nodejs' `http`.

```
await new Promise((resolve, reject) => {
    try {
        app.listen(4873, resolve)
    } catch (e) {
        reject(e)
    }
})
```

Comment (bergundy) [reply scripts/registry.js:79]: Yeah, that's what I usually do, you might need to add `app.on('error', reject)`.

**Comment by bergundy** [comment]
We might need to wait on the server to listen here, see the CI logs: https://github.com/temporalio/sdk-typescript/runs/8139539160?check_suite_focus=true#step:6:15

File: packages/core-bridge/common.js [hunk 3]
```diff
@@ -29,7 +29,7 @@ class PrebuildError extends Error {
   }
 }
 
-function getPrebuiltPath() {
+function getPrebuiltTargetName() {
```
Comment (bergundy) [inline packages/core-bridge/common.js:32]: Thank you for this change! Much better!

File: scripts/publish-to-verdaccio.js [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
-      {
+    try {
+      await spawnNpx(['lerna', 'publish', 'from-package', '--yes', '--registry', 'http://localhost:4873/'], {
         stdio: 'inherit',
-      }
-    );
-    if (status !== 0) {
-      console.error(error);
+        stdout: 'inherit',
+        stderr: 'inherit',
```
Comment (bergundy) [inline scripts/publish-to-verdaccio.js:11]: Yes please :)

**Review by bergundy — APPROVED** [review]

File: scripts/init-from-verdaccio.js [hunk 5]
```diff
@@ -1,28 +1,36 @@
...
+      await spawnNpx(
+        ['@temporalio/create', 'example', '--no-git-init', '--temporalio-version', 'latest', ...initArgs],
+        {
+          stdio: 'inherit',
+          stdout: 'inherit',
+          stderr: 'inherit',
+          cwd: registryDir,
+          env: {
+            ...process.env,
+            NPM_CONFIG_USERCONFIG: npmConfigFile,
```
[the remark this answers is not in the document]
Comment (mjameswh) [reply scripts/init-from-verdaccio.js:25]: The environment variable means that _every_ packages must be requested through Verdaccio. The custom config file makes it possible to only request `@temporalio` packages through Verdaccio.

Requesting a package through Verdaccio is significantly slower than requesting it from NPM... On my local machine, I measure ~12 seconds drop on `init-from-verdaccio` time, when I enable this specific change.

**Pull request merged**